FHEM/90_SIGNALduino_un.pm

```diff
@@ -281,18 +185,82 @@ SIGNALduino_un_Parse($$)
 		#Dispatch($hash, $tscode,undef);
 
 		
-	} else {
-		Log3 $hash, 4, $dummyreturnvalue;
+	}	
+								   
+	##############################################################################################
+	# version 1) message with u..# without development y attribut -> Unknown code u..# , help me!
+	# version 2) message with u..# and development y attribut -> no message for Unknown code
+	##############################################################################################
+	
+	my $value = AttrVal($ioname, "development", "none");
+	my @delevopmentargs = split (",",$value);
+	my $search = "y".$protocol;			# for version 2
+	my $search2 = "u".$protocol;		# for version 1
+
+	Log3 $name, 5, "$name: $ioname AttrVal development: $value" if (defined $value);
+	
+	if (grep{/$search$/ || /$search2$/}@delevopmentargs) {
+		### Help Device + Logfile ###
+	
+		Log3 $name, 4, "$name: $ioname Protocol $protocol ($search) found in AttrVal development!" if (grep{/$search$/}@delevopmentargs);
+		Log3 $name, 4, "$name: $ioname Protocol $protocol ($search2) found in AttrVal development!" if (grep{/$search2$/}@delevopmentargs);
+		
+		my $def;
+		my $devicedef = $name."_".$protocol;
+		$def = $modules{SIGNALduino_un}{defptr}{$devicedef} if(!$def);
+		$modules{SIGNALduino_un}{defptr}{ioname} = $ioname;
+		
+		if(!$def) {
+			Log3 $ioname, 1, "$ioname: $name UNDEFINED sensor " . $devicedef . " detected";
+			return "UNDEFINED $devicedef SIGNALduino_un $devicedef";
+		}
+	
+		my $hash = $def;
+		my $name = $hash->{NAME};
+		my $info = "$bitData | bits=".length($bitData);
+	
+		readingsBeginUpdate($hash);
+		readingsBulkUpdate($hash, "state", "newMSG - ".TimeNow(),0);
```

Comment:
Der Zeitstempel steht doch bereits im Feld für die Zeit. Wozu ist es gut, den auch noch mal in den state zu scrheiben? Wäre es hier nicht besser den Inhalt des readings state mit dem Attribut stateformat zu definieren?

Reply:
👍  dass es mit stateformat auch klappt.

Dass man die Bits im Status nicht reinschreibt ist schon okay. Das wird wohl auch kaum einen Mehrwert bringen die Bits im Status zu sehen oder?

Reply:
Einen Mehrwert bringt dies nicht mit den Bits. Ich habe nun stateformat eingebaut. Wer die Anzeige 
>033540FDBC | 2018-09-19 14:00:47

oder

>UserMSG | 2018-09-19 14:10:12

mag, so löscht man das Attribut Stateformat und erhält fortan nur die HEX Werte beim Empfang und den Status UserMSG nachdem man ein Set für Kommentar absetzte.